### PR TITLE
Fix unknown props warnings

### PR DESCRIPTION
**Pull request by HriBB**
Went through the examples and fixed all unknown props warnings I could find. I am not familiar with the source yet, so there are probably a few more left to omit.

Please let me know if the approach is OK.

### CODE REVIEW

**Comment by HriBB** [comment]
While I was at it, I also fixed `<pre>` issue in examples.

File: src/EventCell.jsx [hunk 1]
```diff
@@ -1,8 +1,20 @@
 import React from 'react';
 import cn from 'classnames';
+import omit from 'lodash/object/omit';
 import dates from './utils/dates';
 import { accessor as get } from './utils/accessors';
 
+const omitProps = [
```
Comment (jquense) [inline src/EventCell.jsx]: it's probably shorter to `pick` the props we need instead of omitting so many

File: src/Month.jsx [hunk 2]
```diff
@@ -28,6 +28,17 @@ let eventsForWeek = (evts, start, end, props) =>
 
 let isSegmentInSlot = (seg, slot) => seg.left <= slot && seg.right >= slot;
 
+const omitProps = [
```
Comment (jquense) [inline src/Month.jsx]: ditto here maybe pick instead?

**Comment by jquense** [comment]
thanks for making an attempt here! a few comments

**Comment by HriBB** [comment]
Yeah I did not like those big arrays as well :) Anyway, found this helpful little package called [`pick-react-known-prop`](https://github.com/tyv/pick-react-known-prop),  which does exactly what we need I guess. Let me know if you like it

**Comment by jquense** [comment]
I think i'd rather not take another dependency here, it doesn't really fix the issue, which is that we should be specific about which props are passed down. Most of these components are internal so we know exactly what props need to be passed to the dom nodes and which don't

**Comment by HriBB** [comment]
OK got it! I Will fix It and only props that are needed. So no funny stuff like `onClick` can come through, right?

**Comment by HriBB** [comment]
Hmm ... I'm wondering if other props are even needed on `<div>`.

`Calendar` receives `className` and `style`
`MonthView` receives only `className`
`EventCell` receives `className`, `style` and `onClick`, which executes `onSelect` prop

I went through all props in examples and only found extra `selected` and `step` [`html-attributes`](https://github.com/alexmingoia/html-attributes/blob/master/lib/html-attributes.js) that are passed to either `Calendar`, `MonthView` or `EventCell`.

Do you maybe know of any other props that are or can be passed to internal `<div>`? If not, then I suggest we simply remove all {...this.props} spreads ;)

**Comment by jquense** [comment]
I think we should remove the spreads, just explicitly pass the props that are needed to the DOM nodes instead. there aren't a lot of them

**Comment by HriBB** [comment]
Done! @jquense please review this.

File: src/Calendar.jsx [hunk 3]
```diff
@@ -390,7 +388,7 @@ let Calendar = React.createClass({
     let ToolbarToRender = components.toolbar || Toolbar
 
     return (
-      <div {...elementProps}
```
Comment (jquense) [inline src/Calendar.jsx]: This is a bit problematic. The user might be adding DOM events or `aria-`/`data-` attributes. Maybe in this case we add a new prop `elementProps` specifically for passing stuff to this div

Comment (HriBB) [inline src/Calendar.jsx]: Something like this?

```
let handlers = [
  'onBlur', 'onChange', 'onClick', 'onContextMenu', 'onCopy',
  'onCut', 'onDoubleClick', 'onDrag', 'onDragEnd', 'onDragEnter',
  'onDragExit', 'onDragLeave', 'onDragOver', 'onDragStart', 'onDrop',
  'onFocus', 'onInput', 'onKeyDown', 'onKeyPress', 'onKeyUp',
  'onMouseDown', 'onMouseEnter', 'onMouseLeave', 'onMouseMove',
  'onMouseOut', 'onMouseOver', 'onMouseUp', 'onPaste',
  'onScroll', 'onSubmit', 'onTouchCancel', 'onTouchEnd', 'onTouchMove',
  'onTouchStart', 'onWheel'
]

let props = {
  'data-id': 1,
  something: 'else',
  invalid: 'prop',
  onClick: function() { console.log('click!'); },
  'data-whatever': 'whateva',
  'aria-invalid': true,
}

function pickValidProps(props) {
  let valid = {}
  for (let p in props) {
    if (handlers.indexOf(p) > -1 || p.startsWith('data-') || p.startsWith('aria-')) {
      valid[p] = props[p]
    }
  }
  return valid
}

console.log('validProps', pickValidProps(props));
```

File: src/EventCell.jsx [hunk 4]
```diff
@@ -24,7 +24,6 @@ let EventCell = React.createClass({
 
     return (
       <div
-        {...props}
```
Comment (HriBB) [inline src/EventCell.jsx]: What about this? Are users able to add custom props here?

File: src/Calendar.jsx (hunk 3 above)
Comment (jquense) [inline src/Calendar.jsx]: no something even simpler. the Calendar has a prop called: `elementProps` which is an object

 then you do: `<div {...this.props.elementProps>` on the calendar DOM node.

File: src/EventCell.jsx (hunk 4 above)
Comment (jquense) [inline src/EventCell.jsx]: I don't think so, I think this is good being removed completely here

File: src/Calendar.jsx (hunk 3 above)
Comment (HriBB) [inline src/Calendar.jsx]: But how are you going to pick which props are allowed?

Comment (jquense) [inline src/Calendar.jsx]: you don't need too the user is in charge of making sure that they are valid

Comment (HriBB) [inline src/Calendar.jsx]: Ohh so you mean something like this?

```
// Calendar.jsx

let Calendar = React.createClass({

  propTypes: {
    elementProps: PropTypes.object,
    ...
  }

  render() {
    let { className, style, elementProps, ...props } = this.props;
    ...
    return (
      <div
        {...elementProps}
        className={cn('rbc-calendar', className, {
          'rbc-rtl': props.rtl
        })}
        style={style}
      >
        ...
      </div>
    );
  }

});
```

And then user passes in `elementProps` in the component

```
// UserCalendar.jsx

import ReactBigCalendar from 'react-big-calendar'

let UserCalendar = React.createClass({

  render() {
    let elementProps = {
      onDrop: function() { ... },
      'data-id': 123
    };
  
    return (
      <ReactBigCalendar elementProps={elementProps} ... />
    );
  }
  
});
```

Comment (jquense) [inline src/Calendar.jsx]: yup :thumbsup:

**Comment by HriBB** [comment]
@jquense ok please review and let me know if there's still anything to fix. We could also add some instructions to `README.md`.

**Comment by jquense** [comment]
Looks good! thanks a lot

**Comment by HriBB** [comment]
Great!

**Pull request merged**